repos/system_upgrade/common/actors/checkpersistentmounts/libraries/checkpersistentmounts.py

```diff
@@ -0,0 +1,57 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import StorageInfo
+
+
+def inhibit_upgrade_due_non_persistent_mount(mountpoint):
+    summary = (
+        'Leapp detected that the {0} mountpoint is mounted, with no corresponding entry in /etc/fstab. '
+        'The upgrade process cannot continue with {0} mounted in non-persistent fashion, '
+        'as Leapp needs this mount to be available after a reboot.'
+    )
+
+    hint = (
+        'Add {0} mount entry to /etc/fstab'
+    )
+
+    reporting.create_report([
+        reporting.Title(
+            'Detected partitions mounted in a non-persistent fashion, preventing a successful in-place upgrade.'
+        ),
+        reporting.Summary(summary.format(mountpoint)),
+        reporting.Remediation(hint=hint.format(mountpoint)),
+        reporting.RelatedResource('file', '/etc/fstab'),
+        reporting.Severity(reporting.Severity.HIGH),
+        reporting.Tags([reporting.Tags.FILESYSTEM]),
+        reporting.Flags([reporting.Flags.INHIBITOR]),
+    ])
+
+
+def mount_entry_with_mountpoint_exists(mount_entries, mountpoint):
+    for mount_entry in mount_entries:
```

Comment:
Nit: L31-L34 are equivalent to `return any(me.mount == mountpoint for me in mount_entries)`